engine/src/main/java/org/terasology/rendering/shader/ShaderParametersCombine.java

```diff
@@ -107,6 +107,12 @@ public void applyParameters(Material program) {
             program.setFloat("outlineThickness", outlineThickness, true);
         }
 
+        //TODO: Other parameters and volumetric fog test case is needed
+        Camera activeCamera = CoreRegistry.get(WorldRenderer.class).getActiveCamera();
```

Comment:
The activeCamera is potentially fetched twice - the first time earlier in the class. Do you mind changing this so that it is fetched at most once? 

Also, since you are working on this file, would you mind fetching the rendering config and the world renderer only once, somewhere at the beginning of applyParameters? Lines such as:

`CoreRegistry.get(Config.class).getRendering().isLocalReflections();`

Should really just be:

`renderingConfig.isLocalReflections();`